packages/blade/src/components/Input/BaseInput/BaseInputTagSlot.web.tsx

```diff
@@ -0,0 +1,201 @@
+
+  // @TODO replace with SSR-friendly hook
+  React.useLayoutEffect(() => {
+    if (!tags) return;
+    if (!tagsContainerRef.current) return;
+
+    const tagElements = tagsContainerRef.current.children;
+    tagsContainerRef.current.classList.add('show');
+    let visibleTagsCount = 0;
+    for (const tagElement of tagElements) {
+      if (isElementVisibleInContainer(tagElement, tagsContainerRef.current)) {
+        visibleTagsCount++;
+      } else {
+        break;
+      }
+    }
+
+    visibleTagsCountRef.current = visibleTagsCount;
+    setInvisibleTagsCount(tags.length - visibleTagsCount);
+    tagsContainerRef.current.classList.remove('show');
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [tags?.length]);
+
+  React.useEffect(() => {
+    if (horizontallyScrollableContainerRef.current && tagRows === '1') {
+      horizontallyScrollableContainerRef.current.scrollTo({
+        top: 0,
+        left: horizontallyScrollableContainerRef.current.scrollWidth,
+        behavior: 'smooth',
+      });
+    }
+
+    if (verticallyScrollableContainerRef.current && (tagRows === '3' || tagRows === 'expandable')) {
+      verticallyScrollableContainerRef.current.scrollTo({
+        top: verticallyScrollableContainerRef.current.scrollHeight,
+        left: 0,
+        behavior: 'smooth',
+      });
+    }
+  }, [tags?.length]);
+
+  return {
+    invisibleTagsCount,
+    tagsContainerRef,
+    horizontallyScrollableContainerRef,
+    verticallyScrollableContainerRef,
+  };
+};
+
+const TAG_HEIGHT = 20;
+const GAP = 8;
+
+const MAX_TAGSLOT_HEIGHT = TAG_HEIGHT * 4 + GAP * 4;
+
+const BaseInputTagSlot = ({
+  tags,
+  setShouldIgnoreBlurAnimation,
+  setFocusOnInput,
+  handleOnClick,
+  tagRows,
+  showAllTags,
+  visibleTagsCountRef,
+  inputWrapperRef,
+}: BaseInputTagSlotProps): React.ReactElement | null => {
+  const {
+    invisibleTagsCount,
+    tagsContainerRef,
+    verticallyScrollableContainerRef,
+    horizontallyScrollableContainerRef,
+  } = useTagsDisplay(tags, visibleTagsCountRef, tagRows);
+
+  if (!tags) {
+    return null;
+  }
+
+  if (tags.length <= 0) {
+    return null;
+  }
+
+  const isMultiline = tagRows === '3' || tagRows === 'expandable';
+
+  // taking 200px as default width when we fail to get width from ref (should be rare). 200px is small enough so tags tend to be shown as +x selected
+  const inputContainerWidth = inputWrapperRef.current?.clientWidth ?? 200;
+  const maxTagContainerWidth: SpacingValueType =
+    tagRows === '3' ? '100%' : `${inputContainerWidth - 100}px`;
+
+  return (
+    <BaseBox
+      ref={horizontallyScrollableContainerRef}
+      paddingLeft="spacing.4"
+      marginY="spacing.2"
+      justifyContent="flex-start"
+      display="flex"
+      flexDirection="column"
+      maxHeight={`${MAX_TAGSLOT_HEIGHT}px`}
+      position="relative"
+      maxWidth="100%"
+      overflowX="auto"
+      // Move to using gap instead of marginLeft on individual tags after RN upgrade
```

Comment:
You can use now

Reply:
https://github.com/razorpay/blade/pull/1459#discussion_r1308224102